keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -0,0 +1,299 @@
+* What kind of processes should be wrapped in a span
+* When to link spans to other traces
+* What kind of data makes for useful attributes
+* How to propagate trace context as proposed above
+
+Having these standards in place will ensure that our tracing instrumentation works well with all backends, and that reviewers have concrete criteria to cross-check PRs against. 
+
+## Graduation requirements
+
+Alpha
+
+- [] Alpha-implementation as described above
+- [] E2e testing of traces
+- [] User-facing documentation
+- [] Tracing must not increase the number of requests to the APIServer
+
+Beta
+
+- [] Security Review, including threat model
+- [] Deployment review including whether the [OC Agent](https://github.com/census-instrumentation/opencensus-service#opencensus-agent) is a required component
+- [] Benchmark kubernetes components using tracing, and determine resource requirements and scaling for any additional required components (e.g. OC Agent).
+
+GA
+
+- [] Versioning for span naming and backwards-compatibility guarantees
+
+## Production Readiness Survey
+
+* Feature enablement and rollback
+  - How can this feature be enabled / disabled in a live cluster?  **Feature-gate**
+  - Can the feature be disabled once it has been enabled (i.e., can we roll
+    back the enablement)?  **Yes, the feature gate can be disabled in all relevant components**
+  - Will enabling / disabling the feature require downtime for the control
+    plane?  **Yes, control-plane components must be restarted with the feature-gate disabled.**
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?  **No, it just requires restarting the kubelet with the feature-gate disabled**
+  - What happens if a cluster with this feature enabled is rolled back? **No Adverse effects** What
+    happens if it is subsequently upgraded again?  **No adverse effects**
+  - Are there tests for this?  **No.  The feature hasn't been developed yet.**
+* Scalability
+  - Will enabling / using the feature result in any new API calls? **No (there was a recent change in the KEP to that effect).**
+    Describe them with their impact keeping in mind the [supported limits][]
+    (e.g. 5000 nodes per cluster, 100 pods/s churn) focusing mostly on:
+     - components listing and/or watching resources they didn't before
+     - API calls that may be triggered by changes of some Kubernetes
+       resources (e.g. update object X based on changes of object Y)
+     - periodic API calls to reconcile state (e.g. periodic fetching state,
+       heartbeats, leader election, etc.)
+  - Will enabling / using the feature result in supporting new API types? **No**
+    How many objects of that type will be supported (and how that translates
+    to limitations for users)?
+  - Will enabling / using the feature result in increasing size or count
+    of the existing API objects?  **Yes.  It adds an annotation to traced objects.**
```

Comment:
Can you add:
- some estimation how big that is (IIUC - it's something like 16B?)
- some short description of how many objects will be traced (e.g. Pods and all "controller" objects, or whatever that shows it's not 1% of objects, rather potentially close to all)

Reply:
It is 32 bytes.  I added something to that effect.